backend/assignments/1/src/main/java/org/example/ExecuteTransaction.java

```diff
@@ -0,0 +1,162 @@
+package org.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
+public class ExecuteTransaction implements Runnable{
+    JsonNode node;
+    CountDownLatch countDownLatch;
+    String coinErrorMsg = "Coin symbol is not valid ";
+    String traderErrorMsg = "Trader wallet address is not valid ";
+    Random rnd;
+    public ExecuteTransaction(JsonNode jsonNode,CountDownLatch countDownLatch){
+        this.node = jsonNode;
+        this.countDownLatch = countDownLatch;
+    }
+    public ExecuteTransaction(){
+
+    }
+
+    @Override
+    public void run() {
+        String type = node.get("type").asText();
+        JsonNode data = node.get("data");
+        String transactionHash = getBlockHash();
+
+        try {
+            switch (type) {
+                case "BUY" -> buyCoin(data);
+
+                case "SELL" -> sellCoin(data);
+
+                case "ADD_VOLUME" -> addVolume(data);
+
+                case "UPDATE_PRICE" -> updatePrice(data);
+
+                default -> Main.logger.logInfo("Invalid type");
+            }
+        }
+        finally {
+            transactionHash = transactionHash+" ";
+            countDownLatch.countDown();
+        }
+    }
+
+    private void buyCoin(JsonNode data){
+        String coinName = data.get("coin").asText();
+        long quantity = data.get("quantity").asLong();
+        String walletAddress = data.get("wallet_address").asText();
+        Traders trader = Main.getTraderbyWallet.get(walletAddress);
+        if (trader == null) {
```

Comment:
if(!trader) can suffice.
Also, using Objects.isNull() for null checks is cleaner